apps/src/applab/applab.js

```diff
@@ -1182,17 +1195,17 @@ Applab.execute = function () {
     }
   }
 
-  if (Applab.makerController) {
-    Applab.makerController
-        .connectAndInitialize(codegen, Applab.JSInterpreter)
-        .catch((error) => {
-          studioApp.displayPlayspaceAlert("error",
-              <div>{`Board connection error: ${error}`}</div>);
+  if (makerEnabled) {
+    connectToMakerBoard()
+        .then(board => {
+          board.installOnInterpreter(codegen, Applab.JSInterpreter);
+          makerCommands.injectBoardController(board);
+          board.once('disconnect', () => studioApp.resetButtonClick());
+          makerBoard = board;
```

Comment:
applab.js _does_ get a little more responsibility with this change, but it feels right to me because all of these things are applab-level concerns, and have to be skipped or handled differently on the setup page (and eventually in game lab):
* Adding board and component controllers to the interpreter.
* Injecting the current board controller into the maker command set.
  * We could do this internally somewhere, but that sort of makes the board controller back into a singleton :cry:
* Resetting the app when the board gets disconnected.